components/modules/spi_master.c

```diff
@@ -296,6 +298,19 @@ static int lspi_host_device( lua_State *L )
   CONFIG_DEVICE_FROM_BOOL_FIELD(halfduplex, SPI_DEVICE_HALFDUPLEX);
   CONFIG_DEVICE_FROM_BOOL_FIELD(clk_as_cs, SPI_DEVICE_CLK_AS_CS);
   lua_settop( L, stack );
+
+  //
+  if (config.spics_io_num >= 0) {
+    // Configure the gpio
+    gpio_config_t cfg;
+    memset(&cfg, 0, sizeof(cfg));
+    cfg.mode = GPIO_MODE_OUTPUT;
+    cfg.pin_bit_mask = 1ULL << config.spics_io_num;
+    if (!no_err(gpio_config (&cfg)) ||
+        !no_err(gpio_set_level (config.spics_io_num, (config.flags & SPI_DEVICE_POSITIVE_CS) ? 0 : 1))) {
+      return luaL_error(L, "failed to configure gpio");
+    }
+  }
```

Comment:
Are these relevant to the RMT work, or should this be separate?

Reply:
This is the in spi PR which got merged into this one. It should go first.